File: backend/src/main/scala/kpn/api/common/search/RouteSearchResult.scala
package kpn.api.common.search

import kpn.api.common.Bounds
import kpn.api.common.RouteScope

case class RouteSearchResult(
  id: Long,
  name: String,
  scopes: Seq[RouteScope],
  distance: Long,
  symbol: Option[String],
  bounds: Option[Bounds],
  routeIds: Seq[Long],
) {
  def toRouteListItem: RouteListItem = {
    RouteListItem(
      id,
      name,
      distance,
      symbol,
      bounds,
      routeIds,
    )
  }
}

File: backend/api/src/main/java/kpn/api/common/search/RouteSearchResult.java
package kpn.api.common.search;

import kpn.api.common.Bounds;
import kpn.api.common.RouteScope;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record RouteSearchResult(
  Long id,
  String name,
  ImmutableList<RouteScope> scopes,
  Long distance,
  Optional<String> symbol,
  Optional<Bounds> bounds,
  ImmutableList<Long> routeIds
) {
}

/* TODO migrate

  def toRouteListItem: RouteListItem = {
    RouteListItem(
      id,
      name,
      distance,
      symbol,
      bounds,
      routeIds,
    )
  }

*/
